rclcpp/include/rclcpp/subscription_options.hpp

```diff
@@ -80,7 +80,8 @@ struct SubscriptionOptionsBase
 
     // An optional QoS which can provide topic_statistics with a stable QoS separate from
     // the subscription's current QoS settings which could be unstable.
-    rclcpp::QoS qos = SystemDefaultsQoS();
+    // Explicitly set the enough depth to avoid missing the statistics messages.
+    rclcpp::QoS qos = SystemDefaultsQoS().keep_last(10);
```

Comment:
@clalancette i think what we would want to fix is default depth for topic statistics publisher. in default, we have 2 types statistics message, the one is `age` and the other is `period`, that means there is already a possibility to miss the message depends on the rmw implementation. i think this is better for user application too, not only for the unit test.

Reply:
@clalancette @fgallegosalido @MiguelCompany thanks for the information.

what do you think fixing the default depth here? i think this would be the best for user application and tests.

Reply:
Yeah, I like it.